src/download/src/lib.rs

```diff
@@ -612,7 +615,12 @@ pub mod hyper_base {
             _ => return Err(format!("unsupported URL scheme: '{}'", url.scheme()).into())
         };
 
-        let mut res = try!(client.get(url.clone()).send()
+        use std::cmp;
+        let byte = cmp::max(resume_from, 0);
+        let header = Range::Bytes(vec![ByteRangeSpec::AllFrom(byte)]);
+        let req = client.get(url.clone()).header(header);
+
+        let mut res = try!(req.send()
                            .chain_err(|| "failed to make network request"));
         if res.status != self::hyper::Ok {
```

Comment:
A request with a `Range` header will return a `PartialContent` (a 206), so this needs to be updated to handle that status.